### ISSUE TITLE
Error: Meteor code must always run within a Fiber.

### ISSUE THREAD

**Issue by pbeets** [matteodem/meteor-easy-search#322]
Trying to use easysearch with elastic and React.

A gist of a simplified version of what I'm doing.
[https://gist.github.com/pbeets/7b79f4222b2197691b83](https://gist.github.com/pbeets/7b79f4222b2197691b83)

When I search the server keeps sending erroring out with the following, and restarting.
```
W20151025-09:52:00.521(-7)? (STDERR) Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
W20151025-09:52:00.522(-7)? (STDERR)     at Object.Meteor._nodeCodeMustBeInFiber (packages/meteor/dynamics_nodejs.js:9:1)
W20151025-09:52:00.522(-7)? (STDERR)     at [object Object]._.extend.get (packages/meteor/dynamics_nodejs.js:21:1)
W20151025-09:52:00.522(-7)? (STDERR)     at Object.collection.(anonymous function) [as find] (packages/matb33_collection-hooks/collection-hooks.js:103:1)
W20151025-09:52:00.523(-7)? (STDERR)     at [object Object]._.extend.find (packages/mongo/collection.js:279:1)
W20151025-09:52:00.523(-7)? (STDERR)     at lib/engine.js:175:43
W20151025-09:52:00.523(-7)? (STDERR)     at respond (/Users/Pbeets/.meteor/packages/easysearch_elasticsearch/.2.0.0.1h2w998++os+web.browser+web.cordova/npm/node_modules/elasticsearch/src/lib/transport.js:277:9)
W20151025-09:52:00.523(-7)? (STDERR)     at checkRespForFailure (/Users/Pbeets/.meteor/packages/easysearch_elasticsearch/.2.0.0.1h2w998++os+web.browser+web.cordova/npm/node_modules/elasticsearch/src/lib/transport.js:219:7)
W20151025-09:52:00.523(-7)? (STDERR)     at HttpConnector.<anonymous> (/Users/Pbeets/.meteor/packages/easysearch_elasticsearch/.2.0.0.1h2w998++os+web.browser+web.cordova/npm/node_modules/elasticsearch/src/lib/connectors/http.js:155:7)
W20151025-09:52:00.523(-7)? (STDERR)     at IncomingMessage.wrapper (/Users/Pbeets/.meteor/packages/easysearch_elasticsearch/.2.0.0.1h2w998++os+web.browser+web.cordova/npm/node_modules/elasticsearch/node_modules/lodash/index.js:3095:19)
W20151025-09:52:00.523(-7)? (STDERR)     at IncomingMessage.emit (events.js:117:20)
W20151025-09:52:00.524(-7)? (STDERR)     at _stream_readable.js:944:16
W20151025-09:52:00.524(-7)? (STDERR)     at process._tickDomainCallback (node.js:492:13)
```

I've ended up using the native client to make the request, return the id's into a subscription on the client side and publish them back to the client manually.

**Comment by matteodem** [comment]
Can you provide a reproduction repo or steps? I don't see any code that has to do with EasySearch in the error stack.

**Comment by streamnsight** [comment]
your engine.js code must use the Meteor.bindEnvironment for the callback to the search function in getSearchCursor
this is what i had done to fix this problem in the elasticsearch pull request

**Comment by pbeets** [comment]
@streamnsight Awesome!

@matteodem I've moved away from using this library though since it does not let you specify a mapping on index create and will overwrite your current mapping every time meteor starts up. Perhaps I might make a pull-request adding this behavior if it is desired.

**Comment by matteodem** [comment]
yeah I'll think the elasticsearch integration needs to be re-thinked. I'm glad to accept a PR that does not clear the mapping, I'm not sure why that happens though since the index is not deleted on meteor startup.

**Comment by matteodem** [comment]
Should be fixed as of 2.0.3.
